docker/duckdb_server.py

```
@@ -0,0 +1,62 @@
+
+        # Convert result to list of dictionaries
+        columns = result.description
+        if columns:
+            column_names = [col[0] for col in columns]
+            rows: Sequence[tuple[Any, ...]] = result.fetchall()
+            return [dict(zip(column_names, row, strict=False)) for row in rows]
+        return []
+
+    except Exception as e:
+        logger.error(f"Error executing query: {str(e)}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)